### ISSUE TITLE
IP range cannot be deleted, but no valid reason is known

### ISSUE THREAD

**Issue by DaanHoogland** [apache/cloudstack#8080]
##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~
UI, API
~~~

##### CLOUDSTACK VERSION

~~~
any <4.19
~~~

##### SUMMARY

trying to delete a zone physical networks and IP ranges need to be deleted first. This can not when an address is in use, but there is no info on which address or what resource uses it.
![image](https://github.com/apache/cloudstack/assets/2486961/12469455-b64d-478d-a210-6775d0a67d0d)

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
the error message would show which IP addresses are in use and preferably by which resources as well.
~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by vishesh92** [comment]
@DaanHoogland The number of IPs in use can be a very high number. Listing all the resources would be difficult. IMO, it would be better to add a button to show all Allocated IPs belonging to the physicalnetwork.

**Comment by DaanHoogland** [comment]
> @DaanHoogland The number of IPs in use can be a very high number. Listing all the resources would be difficult. IMO, it would be better to add a button to show all Allocated IPs belonging to the physicalnetwork.

I would expect this to be a problem only when a user expects to be ready to delete the network. That said, you are right. We could show a maximum of 5 and than a '...'. We can also go for something completely different as long as it is in a way a pointer to a solution for the user. As is it is a bit "computer says no".

**Comment by weizhouapache** [comment]
> @DaanHoogland The number of IPs in use can be a very high number. Listing all the resources would be difficult. IMO, it would be better to add a button to show all Allocated IPs belonging to the physicalnetwork.

looks like a UI improvement
it is already supported to list public ips by physical network.

**Comment by vishesh92** [comment]
@DaanHoogland I have added two buttons to show all IP addresses by physical network & vlanid in this PR: https://github.com/apache/cloudstack/pull/8717

**Comment by DaanHoogland** [comment]
addressed in #8717
